subsys/bluetooth/audio/Kconfig.mctl

```diff
@@ -0,0 +1,84 @@
+# Bluetooth Audio - Media control configuration options
+
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+rsource "Kconfig.mpl"
+
+config MCTL
+	bool "Support for media player control"
+	select EXPERIMENTAL
+	help
+	  Enables support for control of local and remote media players
+	  To enable support for control of a local media player, support for
+	  local media player must be enabled
+
+if MCTL
+
+config MCTL_LOCAL_PLAYER_CONTROL
+	bool "Support for control of local media player"
+	# TODO: Move MPL out of BT susbystem
+	select BT_MPL
+	help
+	  This option enables support for control of a local media player
+
+config MCTL_LOCAL_PLAYER_LOCAL_CONTROL
+	bool "Support for local control of local media player"
+	depends on MCTL_LOCAL_PLAYER_CONTROL
+	help
+	  This option enables support for local application control of local
+	   media players
+
+config MCTL_LOCAL_PLAYER_REMOTE_CONTROL
+	bool "Support for remote control of local media players"
+	depends on MCTL_LOCAL_PLAYER_CONTROL
+	help
+	  This option enables support for remote control of local media
+	   players
+
+config MCTL_LOCAL_PLAYER_REMOTE_CONTROL_BT
```

Comment:
If we keep this for BT-only we can omit the BT postfix (but will need a BT_ prefix). 

If we want to use this for non-BT, then I strongly suggest that e.g. MCS enables this, instead of the other way around.

Reply:
After thinking more, I partly changed my mind, and found that we can at least keep the possibility to allow other protocols.  Therefore the _BT suffix - there is now room to add other protocols without disturbing what is there.

As for which module enables which, it is not obvious to me that the protocol (BT in this case) should be the "center of the world".  But that is a discussion I think we can take separately.  The important thing now is that the configs for the media_proxy, the player and the MCS/MCC are separated, with the corresponding code changes.  Switching the direction of dependencies is then easily done just by modifying Kconfig files.

Reply:
Agreed, this is outside the scope of this PR. However, in that case I'd still remove the _BT suffix, as that is making it more confusing than not at this point. 

Will you create a GH issue to track the potential refactoring of that?

Reply:
Reopened https://github.com/zephyrproject-rtos/zephyr/issues/42523 and added to the description.

Removing the _BT suffix defeats the purpose of the splitting, as it will tie the (generic) remote control option to the Bluetooth protocol.  The option would be to remove the non_BT general option, and put it back when needed, but that hides how the structure is intended to be.
Tried to make less confusing by adding to the help texts.

Reply:
> Removing the _BT suffix defeats the purpose of the splitting,

I disagree. If the media control/proxy is a separate feature from BT, then it should be split regardless. 
However, my opinion still stands that if the media control/proxy is separate from BT, then it should be responsible for enabling/disabling any BT related features, but rather have the BT related features `depend on` or `select` this instead, as we do for `BT_SETTINGS`. 

Having BT specific Kconfigs in a non-BT feature is not the right way to go about this in my opinion. 

Perhaps we should get a 3rd or 4th set on eyes on this as well.

Reply:
I wonder if we may be talking past each other.
Is it the use of "select" you are opposed to?  (That is something to be discussed, but as I said that can easily be fixed later.)
Or is it the _BT suffix for the Bluetooth remote controls?

Reply:
Changed "select" to "depends on"